mchess-master/mchess-master/CBoard.cc

```diff
@@ -241,55 +241,50 @@ void CBoard::find_legal_moves(CMoveList &moves) const
 {
     moves.clear();
 
-    if (m_side_to_move > 0)
+    if (m_side_to_move > 0){
+        int iscaptured=0;
+        int prerun=0;
         for (int i=A1; i<=H8; ++i)
         {
             int8_t piece = m_board[i];
             int j;
-
+            if(iscaptured==1&&prerun==0)
+                moves.clear();
```

Comment:
The whole logic is wrong, you are basically finding all the moves, and pushing the moves into a stack _**moves**_,. See, below, you are literally checking every possible moves. But the problem is, you should first check for all capture moves first, and then check for none capture moves. Only check non capture moves if no capture moves are found.